Python/marshal.c

```diff
@@ -1954,28 +2026,25 @@ marshal_loads_impl(PyObject *module, Py_buffer *bytes, int lazy)
     if (lazy < 0)
         lazy = getenv("LAZY") && *getenv("LAZY");  // TODO: Rename?
     if (lazy) {
-        rf.ctx = PyMem_Malloc(sizeof(struct context));
+        rf.ctx = _PyHydrationContext_new(
+            bytes->obj, s, n, rf.refs);
+        Py_DECREF(rf.refs);
```

Comment:
This DECREF seems misplaced. We're not done with rf.refs, so it should not be DECREF'ed until after the call to read_object() below.

Reply:
The _PyHydrationContext_new increase it so the context is holding a reference to it now.